spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/JpaBaseConfiguration.java

```diff
@@ -195,9 +197,11 @@ static class PersistenceManagedTypesConfiguration {
 		@Bean
 		@Primary
 		@ConditionalOnMissingBean
-		static PersistenceManagedTypes persistenceManagedTypes(BeanFactory beanFactory, ResourceLoader resourceLoader) {
+		static PersistenceManagedTypes persistenceManagedTypes(BeanFactory beanFactory, ResourceLoader resourceLoader,
+				ObjectProvider<ManagedClassNameFilter> managedClassNameFilter) {
 			String[] packagesToScan = getPackagesToScan(beanFactory);
-			return new PersistenceManagedTypesScanner(resourceLoader).scan(packagesToScan);
+			return new PersistenceManagedTypesScanner(resourceLoader, managedClassNameFilter.getIfAvailable())
```

Comment:
I wonder if we should support multiple filters and create a composite. For consistency with exposing actuator endpoints, exclude would take precedence over include. In other words, a class name would have to be matched by every filter for it to be included.

Reply:
> a class name would have to be matched by every filter for it to be included.

It's unintuitive.

Reply:
It's hard to continue a discussion as you haven't described why you think that it's unintuitive. As I said above, this is how the actuator endpoints behave and that seems to be serving us well. I think the consistency would be a good thing.

Reply:
My apologize, as typical spring framework user, I’m expecting one bean available for injection, If multiple beans need to be composed, application should write a primary composite implementation they can take over the control, not hardcoded AND or OR. 
l said unintuitive because I don’t know If you actuator endpoints doing that way if you don’t mention it.

Reply:
@snicoll do you have an opinion here? Do you think there's likely to be a need for multiple filters that we compose automatically or will most users' needs be served by supporting only one. I think I'm now leaning towards only handling one in the auto-configuration.

Reply:
Sorry, I read that and mean to reply and I forgot about it. I honestly don't know but I prefer that the decision is in a central place. If users need to have multiple filters for whatever reason, I'd prefer them to do the gathering rather than us.

Reply:
Sounds like we should stick with a single bean for now. Perhaps some helper factory methods can be added `ManagedClassNameFilter` if composing filters turns out to be a common need? E.g. `ManagedClassNameFilter.allOf(f1,f2,f3)`.